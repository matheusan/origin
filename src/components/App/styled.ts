import styled, { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;

    background-color: ${props => props.theme.colors.blueGray10}
  }
`;

const Container = styled.div`
  width: 100%;
`;

export { GlobalStyle, Container };
