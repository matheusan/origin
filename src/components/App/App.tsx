import * as React from 'react';
import { ThemeProvider } from 'styled-components';

import { originDefaultTheme } from '../../theme/originDefaultTheme';
import * as S from './styled';

import Bar from '../Bar';
import Icon from '../Icon';
import Savings from '../Savings';
import SavingsProvider from '../../redux';

const App: React.SFC = (): JSX.Element => (
  <ThemeProvider theme={originDefaultTheme}>
    <SavingsProvider>
      <S.GlobalStyle />
      <Bar>
        <Icon name="logo" height="32px" />
      </Bar>
      <Savings />
    </SavingsProvider>
  </ThemeProvider>
);

export default App;
