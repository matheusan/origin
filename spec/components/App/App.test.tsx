import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import App from '../../../src/components/App';

describe('App', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
