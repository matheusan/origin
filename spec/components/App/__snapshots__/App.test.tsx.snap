// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "blueGray10": "#F4F8FA",
        "blueGray100": "#CBD5DC",
        "blueGray300": "#8A9CA9",
        "blueGray400": "#708797",
        "blueGray50": "#E9EEF2",
        "blueGray600": "#4D6475",
        "blueGray900": "#1E2A32",
        "brandColorPrimary": "#1B31AB",
        "brandColorSecondary": "#0079FF",
        "neutralWhite": "#FFFFFF",
      },
      "fonts": Object {
        "primary": "Work Sans",
        "secondary": "Rubik",
      },
    }
  }
>
  <Component>
    <GlobalStyleComponent />
    <Bar>
      <Icon
        height="32px"
        name="logo"
      />
    </Bar>
    <Savings />
  </Component>
</ThemeProvider>
`;
